rpm/rpm.go

```diff
@@ -197,7 +199,9 @@ func Probe(
 
 	// We must have reused the connection if we are a self probe!
 	if (probeType == SelfUp || probeType == SelfDown) && !probeTracer.stats.ConnectionReused {
-		panic(!probeTracer.stats.ConnectionReused)
+		if !continueOnConnReuse {
+			panic(!probeTracer.stats.ConnectionReused)
```

Comment:
I am conflicted about this. Having a `panic` here is (I think) a very important part of the test because of the semantics of the protocol. It is a requirement that the self probes execute on a pre-existing connection. If that is not the case then it seems like the integrity of the entire test is up for debate.

What do you think?

Reply:
Yeah, after sleeping on it a bit, I agree. I'm going to drop this particular change from the patch set.

Reply:
Perfect! When we have the consolidated error-reporting feature, we can remove the `panic` and make it an error that the listener can deal with!